lib/engine/step/buy_company.rb

```diff
@@ -14,12 +14,9 @@ def actions(entity)
         return [] if entity.minor?
         return blocks? ? ACTIONS : ACTIONS_NO_PASS if can_buy_company?(entity)
 
-        return PASS if blocks? &&
-                       entity.corporation? &&
-                       @game.abilities(entity,
-                                       time: 'owning_corp_or_turn',
-                                       owner_type: 'corporation',
-                                       strict_time: true)
+        if blocks? && entity.corporation? && @game.abilities(entity, time: 'owning_corp_or_turn', allow_passive: false)
```

Comment:
This line is one of two places where `allow_passive: false` is used. This is ensuring that a company's ability can be used if the company is bought in right before the end of the turn. This doesn't apply for always-on / passive abilities like terrain-discount companies such as Sumitomo Mines Railway in 1889, but could apply for many types of abilities, which is why no ability type is passed in.